src/helpers.js

```diff
@@ -80,6 +80,25 @@ function paramsForAction(action) {
   return results;
 }
 
+// Removes line breaks but preserves paragraphs (double newlines).
+// Preserves line breaks denoted with "\".
+function removeLineBreaks(text) {
+  text = text.replace(/\\\n/gm, "XX");
```

Comment:
cute. :) but rather than introduce special markers, regexes _should_ have the power to do what you want .. did you try it with them? Substituting these special markers has the downsides that (a) if those strings actually exist in the text .. boy that will be weird, and (b) running all these `replace`s over the whole string is a lot of passes. Now, in some situations (like this one) the performance probably isn't going to be a real issue, but it could be, so it's nice to have a faster tool in your toolbox.

Essentially, you want to match a newline that is preceded by any character other than `\`, and not followed by another newline. This might look _something_ like:

    text.replace(/([^\])\r?\n(?!\r|\n)/, function(match) { return match[1]; });